verification/verifier_helpers.go

```diff
@@ -3,15 +3,145 @@ package verification
 import (
 	"crypto/x509"
 	"fmt"
+	nsigner "github.com/notaryproject/notation-core-go/signer"
+	"github.com/notaryproject/notation-go/registry"
 	"strings"
+	"time"
 )
 
 // isCriticalFailure checks whether a VerificationResult fails the entire signature verification workflow.
 // signature verification workflow is considered failed if there is a VerificationResult with "Enforced" as the action but the result was unsuccessful
-func isCriticalFailure(result VerificationResult) bool {
+func isCriticalFailure(result *VerificationResult) bool {
 	return result.Action == Enforced && !result.Success
 }
 
+func (v *Verifier) verifyIntegrity(sigBlob []byte, sigManifest registry.SignatureManifest, outcome *SignatureVerificationOutcome) (*nsigner.SignerInfo, *VerificationResult) {
+	// parse signature
+	var signerInfo *nsigner.SignerInfo
+	var result *VerificationResult
+
+	sigEnv, err := nsigner.NewSignatureEnvelopeFromBytes(sigBlob, nsigner.SignatureMediaType(sigManifest.Blob.MediaType))
+	if err != nil {
+		result = &VerificationResult{
+			Success: false,
+			Error:   fmt.Errorf("unable to parse the digital signature, error : %s", err),
+			Type:    Integrity,
+			Action:  outcome.VerificationLevel.VerificationMap[Integrity],
+		}
+	} else {
+		// verify integrity
+		signerInfo, err = sigEnv.Verify()
+		if err != nil {
+			switch err.(type) {
+			case nsigner.SignatureNotFoundError, nsigner.MalformedSignatureError, nsigner.SignatureIntegrityError:
+				result = &VerificationResult{
+					Success: false,
+					Error:   err,
+					Type:    Integrity,
+					Action:  outcome.VerificationLevel.VerificationMap[Integrity],
+				}
+			default:
+				// unexpected error
+				result = &VerificationResult{
+					Success: false,
+					Error:   ErrorVerificationInconclusive{msg: err.Error()},
+					Type:    Integrity,
+					Action:  outcome.VerificationLevel.VerificationMap[Integrity],
+				}
+			}
+		} else {
+			// integrity has been verified successfully
+			result = &VerificationResult{
+				Success: true,
+				Type:    Integrity,
+				Action:  outcome.VerificationLevel.VerificationMap[Integrity],
+			}
+		}
+	}
+
+	return signerInfo, result
+}
+
+func (v *Verifier) verifyAuthenticity(trustStorePrefix TrustStorePrefix, trustPolicy *TrustPolicy, outcome *SignatureVerificationOutcome) *VerificationResult {
+	// verify authenticity
+	trustStores, err := loadX509TrustStores(trustPolicy, "testdata/trust-store") // TODO get trust store path from dir structure PR
```

Comment:
This is a big TODO 💀

Reply:
Dir Structure was not implemented when I raised this PR. I will integrate with https://github.com/notaryproject/notation-go/tree/main/dir and remove these TODOs shortly